torchtune/modules/transforms/tokenizers/_hf_tokenizer.py

```
@@ -146,3 +151,158 @@ def decode(self, token_ids: list[int]) -> str:
+
+
+class HuggingFaceModelTokenizer(ModelTokenizer):
+    """
+    A wrapper around Hugging Face model specific tokenizers.
+    This can be used to build a tokenizer with specified model chat template.
+
+    This class will create HuggingFaceBaseTokenizer and load all required tokenizer files.
+    Then, it will load all special tokens and chat template from tokenizer config file.
+
+    It can be used to tokenize messages with correct chat template, and it eliminates the requirement of
+    the specific ModelTokenizer and custom PromptTemplate.
+
+    Args:
+        tokenizer_json_path (str): Path to tokenizer.json file
+        tokenizer_config_json_path (Optional[str]): Path to tokenizer_config.json file. Default: None
+        generation_config_path (Optional[str]): Path to generation_config.json file.
+            Default: None
+        truncation_type (str): type of truncation to apply, either "left" or "right".
+            Default is "right".
+    """
+
+    def __init__(
+        self,
+        tokenizer_json_path: str,
+        *,
+        tokenizer_config_json_path: Optional[str] = None,
+        generation_config_path: Optional[str] = None,
+        truncation_type: str = "right",
+    ):
+        self.base_tokenizer = HuggingFaceBaseTokenizer(
+            tokenizer_json_path=tokenizer_json_path,
+            tokenizer_config_json_path=tokenizer_config_json_path,
+            generation_config_path=generation_config_path,
+        )
+
+        # Contents of the tokenizer_config.json
+        config = self.base_tokenizer.config
+
+        self.special_tokens = _infer_special_tokens_from_hf_config(config)
+        self.top_level_variables = self.extract_top_level_variables(config)
+
+        _env = jinja2.Environment(undefined=StrictUndefined)
+
+        # It is used sometimes in HF chat_templates
+        _env.globals["raise_exception"] = self._raise_helper
+
+        self.template = _env.from_string(config["chat_template"])
```

There are cases where this field is not present, right? E.g. for google/gemma-2-2b (it's only present in the instruct-tuned version). I think for cases like these we should raise an explicit error here.

---

I know this is merged, but LLaMA-4 looks like it has a separate file - `chat_template.jinja` - rather than containing it in the config. This might be another pattern that needs supporting.